include/aws/http/http.h

```diff
@@ -59,6 +59,7 @@ enum aws_http_errors {
     AWS_ERROR_HTTP_WEBSOCKET_PROTOCOL_ERROR,
     AWS_ERROR_HTTP_MANUAL_WRITE_NOT_ENABLED,
     AWS_ERROR_HTTP_MANUAL_WRITE_HAS_COMPLETED,
+    AWS_ERROR_HTTP_REQUEST_IDLE_TIMEOUT,
```

Comment:
ugh, I'm super on the fence about this name.
we don't have any other errors that refer to "REQUEST"
and it's not the "request" that times out, it's the "response"

I get that people often refer to the whole operation as "request" (when "stream" or "exchange" would be more accurate)

`AWS_ERROR_HTTP_RESPONSE_TIMEOUT`?